services/github/github-tag.tester.js

```diff
@@ -28,6 +28,10 @@ t.create('Tag (repo not found)')
   .get('/v/tag/badges/helmets.json')
   .expectBadge({ label: 'tag', message: 'repo not found' })
 
+t.create('Tag (filter by prefix + trim prefix at slash)')
+  .get('/v/tag/ros/rosdistro.json?subpackage=galactic/&trimPrefixAtSlash')
```

Comment:
As I was reading through the diff again I was starting to feel like I'd be more comfortable with a greater number of tests, and I definitely feel that way after getting to this test, which is using the no-longer valid/existent query parameter name.

I think it would be helpful to try to move more of the logic out of the `handle` function(s) (which are more difficult to test) into standalone functions (either class members or free functions) that can be directly unit tested and consumed by `handle` and/or `fetch` functions as needed

Reply:
I fixed that test. I had assumed that `Joi.date().required()` meant that the string had to be only a date, not just include one. Since all that repo's tags are something like `galactic/2022-10-04` it matched on all tags, which is disappointing because originally I had made a regex to just match the date but switched to `Joi.date` instead on feedback. Now It checks for a date and a prefix of `20`.

I just reviewed the code and I don't know how much more I can break up. I could maybe make a function of splitting the prefix but when you want to trim but I don't see how much more the code could be split. If you have some pointers I would gladly take them.

Thanks for all the time you have spent reviewing this!